jwst/group_scale/tests/test_group_scale.py

```diff
@@ -0,0 +1,144 @@
+    datmod = make_rampmodel(2, 2, 4, 2048, 2048)
+    output = GroupScaleStep.call(datmod)
+    
+    # Assert that the step completed
+    assert(output.meta.cal_step.group_scale == 'COMPLETE')
+    
+    # This assertion doesn't verify for correct output, 
+    # it just checks that the correction ran and that the data array 
+    # outputs are different than the inputs as requested in the document.
+    assert not np.array_equal(output.data, datmod.data)
+
+
+def test_nframes_is_none(make_rampmodel):
+    """Make sure step is skipped if nframes is None
+    """
+    datmod = make_rampmodel(2, None, 4, 2048, 2048)
+    output = GroupScaleStep.call(datmod)
+
+    assert(output.meta.cal_step.group_scale == 'SKIPPED')
+
+
+def test_nframes_is_power_of_two(make_rampmodel):
+    """When frame_divisor is None, the correction will skip if
+    nframes is a power of 2.
+    """
+    datmod = make_rampmodel(2, 4, None, 2048, 2048)
+    output = GroupScaleStep.call(datmod)
+
+    assert(output.meta.cal_step.group_scale == 'SKIPPED')
+
+
+def test_nframes_is_not_power_of_two(make_rampmodel):
+    """When frame_divisor is None, then do_correction will be applied if
+    nframes is not a power of two but will be skip because if nframes or
+    frame_divisor is none.
+    """
+    datmod = make_rampmodel(2, 3, None, 2048, 2048)
+    output = GroupScaleStep.call(datmod)
+
+    assert(output.meta.cal_step.group_scale == 'SKIPPED')
+
+def test_scale_value(make_rampmodel):
+    """Compare the ratio of the FRMDIVSR/NFRAMES from the data model input and
+    compare to the output of the pipeline.
+    """
+
+    datmod = make_rampmodel(2, 2, 4, 2048, 2048)
+
+    # Calculate the scale based off of the input.
+    scale = datmod.meta.exposure.frame_divisor/datmod.meta.exposure.nframes
+    
+    output = GroupScaleStep.call(datmod)
+
+    scale_from_data = np.unique(output.data/datmod.data)
+
+    # Since the scale value is applied uniformly to the array, if we divide the output
+    # by the input then we should get a single unique value (ie the scale) calculated
+    # by the pipeline. 
+    assert(len(scale_from_data) == 1)
+
+    # Make sure the scale calculated manually from the data model aboved matched what the
+    # pipeline calculated.
+    assert(scale == scale_from_data[0])
+
+
+# Make this a test eventually. The pipeline step nor the caldetector1 pipeline
+# code checks for instrument specific mode information. We want to make sure 
+# in the future that the group_scale calculation is only run on IRS2 mode data 
+# as defined in the confluence table for this step. Here we set the read pattern 
+# mode to 'NRSIRS2RAPID' to make sure it fails on purpose.
+#
+# @pytest.mark.xfail
+# def test_correction_only_for_irs2(make_rampmodel):
+#     """Make sure pipeline doesnt run correction on data that isnt IRS2
+#     """
+#     datmod = make_rampmodel(2, 2, 4, 2048, 2048)
+#     datmod.meta.readpatt = 'NRSIRS2RAPID'
+#     output = GroupScaleStep.call(datmod)
```

Comment:
I think you can uncomment the test since you've marked it as xfail.  Perhaps do
```
@pytest.mark.xfail(reason='Not implemented yet")
```
or something more descriptive.

Reply:
Hey @jdavies-st I have been looking at the source code some more and from what I am noticing the code is written to be flexible and work for any `instrument` regardless of the `readpatt`. This seems to be handled in the reference files or are assigned in the headers by CRDS (outside of the scope of the JWST testing group). So I think I will just go ahead and remove it!